modin/test/storage_formats/pandas/test_internals.py

```diff
@@ -433,3 +486,94 @@ def test_virtual_partition_dup_object_ref():
     ), "Test setup did not contain duplicate objects"
     # The below call to wait() should not crash
     partition.wait()
+
+
+__test_reorder_labels_cache_axis_positions = [
+    pytest.param(lambda index: None, id="no_reordering"),
+    pytest.param(lambda index: np.arange(len(index) - 1, -1, -1), id="reordering_only"),
+    pytest.param(
+        lambda index: [0, 1, 2, len(index) - 3, len(index) - 2, len(index) - 1],
+        id="projection_only",
+    ),
+    pytest.param(
+        lambda index: np.repeat(np.arange(len(index)), repeats=3), id="size_grow"
+    ),
+]
+
+
+@pytest.mark.parametrize("row_positions", __test_reorder_labels_cache_axis_positions)
+@pytest.mark.parametrize("col_positions", __test_reorder_labels_cache_axis_positions)
+@pytest.mark.parametrize(
+    "partitioning_scheme",
+    [
+        pytest.param(
+            lambda df: {
+                "row_lengths": [df.shape[0]],
+                "column_widths": [df.shape[1]],
+            },
+            id="single_partition",
+        ),
+        pytest.param(
+            lambda df: {
+                "row_lengths": [
+                    min(32, df.shape[0]),
+                    max(0, df.shape[0] - 32),
```

Comment:
Why not just use `[df.shape[0], 0]`? It's just not very convenient to read such a construction (with `min` and `max`).

Reply:
This is intended to test the `[32, N]` case, where there are 2 non-zero unbalanced partitions. The N here is computed as `len(df) - 32`, for default `test_data_values` it's okay to not use `max(0, N)` construction as the `df` length is way bigger than 32. However, for the sake of compatibility with possibly smaller datasets, we have to add this `max` expression so we woudln't break the splitting function with negative lengths.